packages/ui/src/components/Modal/Dialog.tsx

```diff
@@ -0,0 +1,151 @@
+import styled from '@emotion/styled'
+import type { KeyboardEventHandler, MouseEventHandler } from 'react'
+import { useCallback, useEffect, useRef } from 'react'
+import { createPortal } from 'react-dom'
+import { MODAL_PLACEMENT, MODAL_WIDTH } from './constants'
+import type { DialogProps, ModalPlacement, ModalSize } from './types'
+
+const StyledBackdrop = styled.div<{ 'data-open': boolean }>`
+  position: fixed;
+  top: 0;
+  right: 0;
+  height: 0;
+  width: 0;
+  overflow: hidden;
+
+  &[data-open='true'] {
+    overflow: auto;
+    padding: ${({ theme }) => theme.space['2']};
+    display: flex;
+    background-color: ${({ theme }) => theme.colors.overlay};
+    bottom: 0;
+    left: 0;
+    height: 100%;
+    width: 100%;
+    opacity: 1;
+  }
+`
+
+type StyledDialogProps = {
+  'data-size': ModalSize
+  'data-placement': ModalPlacement
+}
+
+const StyledDialog = styled.dialog<StyledDialogProps>`
```

Comment:
Support is relatively recent on Safari, do we have a way to check if this is going to be ok with our user base ?

Reply:
@chambo-e there is 93% of the global users that are compatible with this, what is our minimum in order to use it?

Source: https://caniuse.com/?search=dialog